hypervisor/arch/x86/guest/mptable.c

```diff
@@ -292,12 +292,16 @@ int mptable_build(struct vm *vm)
 	char                    *curraddr;
 	struct mpcth            *mpch;
 	struct mpfps            *mpfp;
+	size_t			mptable_length, table_length;
 
 	startaddr = (char *)GPA2HVA(vm, MPTABLE_BASE);
+
+	table_length = vm->vm_desc->mptable->mpch.base_table_length;
+	mptable_length = sizeof(struct mpfps) + table_length;
 	/* Copy mptable info into guest memory */
-	(void)memcpy_s((void *)startaddr, sizeof(struct mptable_info),
+	(void)memcpy_s((void *)startaddr, MPTABLE_MAX_LENGTH,
```

Comment:
Why MPTABLE_MAX_LENGTH is (65536 + 16)U ?
And I am not sure if (65536 + 16)U is an valid expression.

Reply:
sainath fixed it.